knative-operator/pkg/controller/knativekafka/replicastransform.go

```diff
@@ -0,0 +1,46 @@
+package knativekafka
+
+import (
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// replicasTransform makes kafka-ch-dispatcher keep its current replica count set by controller
+// based on vendor/knative.dev/operator/pkg/reconciler/knativeeventing/common/replicasenvvarstransform.go
+func replicasTransform(client mf.Client) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() == "Deployment" && u.GetName() == "kafka-ch-dispatcher" {
+			currentU, err := client.Get(u)
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+			apply := &appsv1.Deployment{}
+			if err := scheme.Scheme.Convert(u, apply, nil); err != nil {
+				return err
+			}
+
+			current := &appsv1.Deployment{}
+			if err := scheme.Scheme.Convert(currentU, current, nil); err != nil {
+				return err
+			}
+
+			// Keep the existing number of replicas in the cluster for the deployment
+			apply.Spec.Replicas = current.Spec.Replicas
+
```

Comment:
Do we need to take care of the `ENV_VARS` too ? 

See:
* https://github.com/knative/operator/issues/393
* https://github.com/knative/operator/pull/394

/cc @lionelvillard

Reply:
I don't think so, looking at https://github.com/knative-sandbox/eventing-kafka/blob/release-0.20/pkg/channel/consolidated/reconciler/controller/kafkachannel.go#L314  , it should accept the Deployment as is if created by the Operator, as long as the container image matches.